ci/ray_ci/automation/docker_tags_lib.py

```diff
@@ -0,0 +1,253 @@
+    response = requests.get(auth_url)
+    if response.status_code != 200:
+        raise AuthTokenException(f"Docker. Error code: {response.status_code}")
+    token = response.json().get("token", None)
+    return token
+
+
+def get_docker_hub_auth_token(username: str, password: str) -> Optional[str]:
+    """Retrieve Docker Hub auth token."""
+
+    url = "https://hub.docker.com/v2/users/login"
+    json_body = {
+        "username": username,
+        "password": password,
+    }
+    headers = {"Content-Type": "application/json"}
+    response = requests.post(url, headers=headers, json=json_body)
+    if response.status_code != 200:
+        raise AuthTokenException(f"Docker Hub. Error code: {response.status_code}")
+    return response.json().get("token", None)
+
+
+def _get_git_log(n_days: int = 30) -> str:
+    return subprocess.check_output(
+        [
+            "git",
+            "log",
+            f"--until='{n_days} days ago'",
+            "--pretty=format:%H",
+        ],
+        text=True,
+    )
+
+
+def list_recent_commit_short_shas(n_days: int = 30) -> List[str]:
+    """
+    Get list of recent commit SHAs (short version) on ray master branch.
+
+    Args:
+        n_days: Number of days to go back in git log.
+
+    Returns:
+        List of recent commit SHAs (short version).
+    """
+    commit_shas = _get_git_log(n_days=n_days)
+    short_commit_shas = [
+        commit_sha[:SHA_LENGTH] for commit_sha in commit_shas.split("\n") if commit_sha
+    ]
+    return short_commit_shas
+
+
+def get_config_docker_oci(tag: str, namespace: str, repository: str):
+    """Get Docker image config from tag using OCI API."""
+    token = get_docker_auth_token(namespace, repository)
+
+    # Pull image manifest to get config digest
+    headers = {
+        "Authorization": f"Bearer {token}",
+        "Accept": "application/vnd.docker.distribution.manifest.v2+json",
+    }
+    image_manifest_url = (
+        f"https://registry-1.docker.io/v2/{namespace}/{repository}/manifests/{tag}"
+    )
+    response = requests.get(image_manifest_url, headers=headers)
+    if response.status_code != 200:
+        raise RetrieveImageConfigException("image manifest.")
+    config_blob_digest = response.json()["config"]["digest"]
+
+    # Pull image config
+    config_blob_url = f"https://registry-1.docker.io/v2/{namespace}/{repository}/blobs/{config_blob_digest}"  # noqa E501
+    config_headers = {
+        "Authorization": f"Bearer {token}",
+        "Accept": "application/vnd.docker.container.image.v1+json",
+    }
+    response = requests.get(config_blob_url, headers=config_headers)
+    if response.status_code != 200:
+        raise RetrieveImageConfigException("image config.")
+    return response.json()
+
+
+def get_image_creation_time(tag: str) -> datetime:
+    """
+    Get Docker image creation time from tag image config.
+
+    Args:
+        tag: Docker tag name
+
+    Returns:
+        Datetime object of image creation time.
+    """
```

Comment:
and also remove the docs for functions that are obvious from the function name?

Reply:
done!